vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java

```diff
@@ -72,57 +77,72 @@
+                  .collect(Collectors.groupingBy(e -> StringUtils.substringBefore(e.getKey(), DOT),
+                      mapping(e -> Map.entry(StringUtils.substringAfter(e.getKey(), DOT), e.getValue()),
+                          Collectors.collectingAndThen(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue),
+                                  factoryCreator::apply))));
+    }
+
     /**
      * Send the data to the provided topic with no key or partition.
-     * @param data the data to send
-     * @param topic the topic name
-     * @throws InterruptedException if the current thread was interrupted while waiting
-     * @throws ExecutionException if the computation threw an exception
-     * @throws TimeoutException if the wait timed out
+     * @param data                  The data to send
+     * @param producerKey           The key of the producer configuration
+     * @param topic                 The topic name
+     * @throws InterruptedException If the current thread was interrupted while waiting
+     * @throws ExecutionException   If the computation threw an exception
+     * @throws TimeoutException     If the wait timed out
      */
-    @When("I send data `$data` to Kafka topic `$topic`")
-    public void sendData(String data, String topic) throws InterruptedException, ExecutionException, TimeoutException
+    @When("I send data `$data` to `$producerKey` Kafka topic `$topic`")
+    public void sendData(String data, String producerKey, String topic) throws InterruptedException, ExecutionException,
+        TimeoutException
     {
-        kafkaTemplate.send(topic, data).get(WAIT_TIMEOUT_IN_MINUTES, TimeUnit.MINUTES);
+        kafkaTemplates.get(producerKey).send(topic, data).get(WAIT_TIMEOUT_IN_MINUTES, TimeUnit.MINUTES);
     }
 
     /**
      * Starts the Kafka consumer with the provided configuration to listen the specified topics. The consumer must be
      * stopped when it's not needed.
      *
-     * @param topics the comma-separated set of topics to listen
+     * @param consumerKey The key of the producer configuration
+     * @param topics      The comma-separated set of topics to listen
      */
-    @When("I start consuming messages from Kafka topics `$topics`")
-    public void startKafkaListener(Set<String> topics)
+    @When("I start consuming messages from `$consumerKey` Kafka topics `$topics`")
+    public void startKafkaListener(String consumerKey, Set<String> topics)
     {
-        stopListener(false);
+        String listenerKey = listenerKey(consumerKey);
+        stopListener(listenerKey, false);
         BlockingQueue<String> messageQueue = new LinkedBlockingDeque<>();
-        testContext.put(MESSAGES_KEY, messageQueue);
-        ContainerProperties containerProperties = new ContainerProperties(topics.toArray(new String[0]));
+        testContext.put(messagesKey(consumerKey), messageQueue);
+        var containerProperties = new ContainerProperties(topics.toArray(new String[0]));
```

Comment:
I don't think it makes sense to follow SQ warnings like this in the production code